lms/djangoapps/email_marketing/signals.py

```diff
@@ -35,35 +35,39 @@ def handle_enroll_status_change(sender, event=None, user=None, mode=None, course
     if not email_config.enabled or not event or not user or not mode or not course_id:
         return
 
+    # skip tracking (un)enrolls if simulated cost=0
+    if email_config.sailthru_enroll_cost == 0:
+        return
+
     request = crum.get_current_request()
     if not request:
         return
 
+    # get string course_id serializable to send through celery
+    course_id_string = unicode(course_id)
+
     # figure out course url
-    course_url = _build_course_url(request, course_id.to_deprecated_string())
+    course_url = _build_course_url(request, course_id_string, email_config)
 
     # pass event to email_marketing.tasks
     update_course_enrollment.delay(user.email, course_url, event, mode,
-                                   unit_cost=cost, course_id=course_id, currency=currency,
+                                   course_id=course_id_string,
                                    message_id=request.COOKIES.get('sailthru_bid'))
 
 
-def _build_course_url(request, course_id):
+def _build_course_url(request, course_id, email_config):
     """
     Build a course url from a course id and the host from the current request
+    or use override in config
     :param request:
     :param course_id:
     :return:
     """
-    host = request.get_host()
-    # hack for integration testing since Sailthru rejects urls with localhost
-    if host.startswith('localhost'):
-        host = 'courses.edx.org'
-    return '{scheme}://{host}/courses/{course}/info'.format(
-        scheme=request.scheme,
-        host=host,
-        course=course_id
-    )
+    if email_config.sailthru_lms_url_override:
+        return '{}/courses/{}/info'.format(email_config.sailthru_lms_url_override, course_id)
+    else:
+        host = request.get_host()
+        return '{}://{}/courses/{}/info'.format(request.scheme, host, course_id)
```

Comment:
It would be good to use the django `reverse` method to construct urls that are hosted within this app, so that you don't have to worry about those urls falling out of sync with the app in the future.